### PR TITLE
Add asynchronous MXRManager initialization

### PR DESCRIPTION
**Pull request by adrenak**
[A doc discussing the problem with old init along with Upgrade & Testing Guide is here](https://app.clickup.com/t/85zt1ej03)
- InitAsync creates an IMXRSystem object and waits until it's available and initialized with state
- Also Added IMXRSystem.OnAvailabilityChange to provide availability change events
- Deprecated on synchronous initialisation as it requires manually checking if the system is ready, making the MXRManager a more complete init solution

### CODE REVIEW

File: Assets/MXR.SDK/Runtime/MXRManager.cs [hunk 1]
```diff
@@ -22,39 +27,102 @@ public static class MXRManager {
...
 
+        [Obsolete("This boolean property was used with the Init method which is being deprecated. " +
+        "Use InitAsync instead.", false)]
         /// <summary>
         /// Whether <see cref="MXRManager"/> has been initialized
         /// </summary>
         public static bool IsInitialized { get; private set; } = false;
 
+        /// <summary>
+        /// Auto-initializes <see cref="System"/> with appropriate implemenration
```
Comment (Aleyum) [inline Assets/MXR.SDK/Runtime/MXRManager.cs:45]: * implementation

**Review by Aleyum — CHANGES_REQUESTED** [review]

**Comment by Aleyum** [comment]
All good from me

File: Assets/MXR.SDK/Runtime/MXRManager.cs [hunk 2]
```diff
@@ -7,6 +10,8 @@ namespace MXR.SDK {
     /// an API to get/set/update the HomeScreenState.
     /// </summary>
     public static class MXRManager {
+        const string TAG = "[MXRManager";
```
Comment (johnmanagexr) [inline Assets/MXR.SDK/Runtime/MXRManager.cs:13]: I think you want [MXRManager] ?

File: Assets/MXR.SDK/Runtime/MXRManager.cs [hunk 3]
```diff
@@ -22,39 +27,102 @@ public static class MXRManager {
...
 
+        /// <summary>
+        /// Auto-initializes <see cref="System"/> with appropriate implementation
+        /// based on the runtime environment and awaits until the system is available 
+        /// for the first time.
+        /// </summary>
+        /// <param name="system"></param>
+        /// <returns>The <see cref="IMXRSystem"/> instance in MXRManager</returns>
+        async public static Task<IMXRSystem> InitAsync(IMXRSystem system = null) {
+#pragma warning disable 0618
```
Comment (johnmanagexr) [inline Assets/MXR.SDK/Runtime/MXRManager.cs:52]: Can you add a comment explaining why this is ok to disable and restore?

**Review by johnmanagexr — CHANGES_REQUESTED** [review]
Can you link a clickup ticket and add internal testing steps to it? Let's get this testes before it goes in!

File: Assets/MXR.SDK/Runtime/MXRManager.cs (hunk 3 above)
Comment (adrenak) [reply Assets/MXR.SDK/Runtime/MXRManager.cs:57]: Added!

File: Assets/MXR.SDK/Runtime/MXRManager.cs (hunk 2 above)
Comment (adrenak) [reply Assets/MXR.SDK/Runtime/MXRManager.cs]: Fixed

**Comment by adrenak** [comment]
Created a clickup task [here](https://app.clickup.com/t/85zt1ej03) with explainer and testing steps. It's a bit of a doc too explaining the motivations behind the new init method and the problems faced.

**Review by johnmanagexr — APPROVED** [review]

**Review by Aleyum — APPROVED** [review]

**Pull request merged**